edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.CloudProxy/ServiceProxy.cs

```diff
@@ -175,6 +177,61 @@ public static void BadRequestResult(string id, HttpStatusCode statusCode)
             }
         }
 
+        /// <summary>
+        /// This is a helper class used to pull down the devices/modules in scope for every nested Edge Device.
+        /// It maintains a queue of ServiceIdentitiesIterator, each representating a child Edge device
+        /// in the subtree, and uses these iterators to perform breadth-first-expansion on the tree.
+        /// </summary>
+        class NestedServiceIdentitiesIterator : IServiceIdentitiesIterator
+        {
+            IDeviceScopeApiClient rootClient;
+            Queue<IServiceIdentitiesIterator> remainingEdgeNodes;
```

Comment:
>IServiceIdentitiesIterator [](start = 18, length = 26)

From David: Maybe instead of having a queue of iterators that deals with the continuation token, just maintain a queue of ApiClients.  Then every time we need to use the continuation token, enqueue a new ApiClient.